DuckDuckGo/TabViewController.swift

```diff
@@ -167,6 +168,9 @@ class TabViewController: UIViewController {
     var mostRecentAutoPreviewDownloadID: UUID?
     private var blobDownloadTargetFrame: WKFrameInfo?
 
+    // Recent request's URL if its WKNavigationAction had shouldPerformDownload set to true
+    private var recentNavigationActionShouldPerformDownloadURL: URL?
```

Comment:
This is used to hold and pass information from `WKNavigationAction`'s  `shouldPerformDownload` read in: 
- `func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void)`

to be used for decision logic later in:
- `func webView(_ webView: WKWebView, decidePolicyFor navigationResponse: WKNavigationResponse, decisionHandler: @escaping (WKNavigationResponsePolicy) -> Void)`

and
- `func webView(_ webView: WKWebView, navigationResponse: WKNavigationResponse, didBecome download: WKDownload)`